<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_sql_database_invalid_requested_service_objective_name

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- System
- System0
- System1
- System2
- System3
- System4
- System2L
- System3L
- System4L
- Free
- Basic
- S0
- S1
- S2
- S3
- S4
- S6
- S7
- S9
- S12
- P1
- P2
- P3
- P4
- P6
- P11
- P15
- PRS1
- PRS2
- PRS4
- PRS6
- DW100
- DW200
- DW300
- DW400
- DW500
- DW600
- DW1000
- DW1200
- DW1000c
- DW1500
- DW1500c
- DW2000
- DW2000c
- DW3000
- DW2500c
- DW3000c
- DW6000
- DW5000c
- DW6000c
- DW7500c
- DW10000c
- DW15000c
- DW30000c
- DS100
- DS200
- DS300
- DS400
- DS500
- DS600
- DS1000
- DS1200
- DS1500
- DS2000
- ElasticPool

## Example

```hcl
resource "azurerm_sql_database" "foo" {
  requested_service_objective_name = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as requested_service_objective_name (azurerm_sql_database_invalid_requested_service_objective_name)

  on template.tf line 2:
  2:   requested_service_objective_name = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_sql_database_invalid_requested_service_objective_name.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/databases.json
